internal/ingress/controller/parser/parser.go

```diff
@@ -154,13 +154,25 @@ func tcpIngressToNetworkingTLS(tls []configurationv1beta1.IngressTLS) []networki
 func findPort(svc *corev1.Service, wantPort kongstate.PortDef) (*corev1.ServicePort, error) {
 	switch wantPort.Mode {
 	case kongstate.PortModeByNumber:
+		// ExternalName Services have no port declaration of their own
+		// We must assume that the user-requested port is valid and construct a ServicePort from it
+		if svc.Spec.Type == corev1.ServiceTypeExternalName {
+			return &corev1.ServicePort{
+				Protocol:   "TCP",
```

Comment:
```suggestion
```

Reply:
@rainest

> I think it's fine to leave the stub configuration here since we'll ignore it when creating the Kong configuration later anyway

In such case we should be able to remove the protocol field whatsoever, right?

Reply:
Maybe. Tests pass with it gone, but our coverage isn't great, and my concern is that there's something elsewhere that will try to make use of it and panic if it's nil. The original 0.8 code did include it, so IMO safest thing is to just let that persist. However, grep doesn't show anything in our code that looks like it could use it.